src/level_gen.cpp

```diff
@@ -176,6 +176,11 @@ void createPersistentEntities(Engine &ctx)
             other_agents.e[out_idx++] = other_agent;
         }
     }
+
```

Comment:
Can you move these queries to Sim::Sim()? They don't really belong in level setup code.